google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -3343,6 +3441,43 @@ private void createSessions(final int sessionCount, boolean distributeOverChanne
     }
   }
 
+  class MultiplexedSessionConsumer implements SessionConsumer {
+
+    @Override
+    public void onSessionReady(SessionImpl sessionImpl) {
+      final SettableFuture settableFuture = SettableFuture.create();
+      final MultiplexedSession newSession = new MultiplexedSession(sessionImpl);
+      settableFuture.set(newSession);
+
+      synchronized (lock) {
+        if (multiplexedSessionFuture.get() != null && multiplexedSessionFuture.get().isDone()) {
+          final MultiplexedSession oldSession = getMultiplexedSession().get();
+          logger.log(
+              Level.INFO,
+              String.format(
+                  "Removed Multiplexed Session => %s created at => %s and",
+                  oldSession.getName(), oldSession.getDelegate().getCreateTime()));
+          if (multiplexedSessionRemovedListener != null) {
+            multiplexedSessionRemovedListener.apply(oldSession);
+          }
+        }
+        SettableApiFuture settableApiFuture = SettableApiFuture.create();
+        settableApiFuture.set(
+            new MultiplexedSessionFuture(settableFuture, tracer.getCurrentSpan()));
```

Comment:
Why is `tracer.getCurrentSpan()` passed in to this constructor? The current span is the span that has been set on the background thread that created the session. Is that a span that we want to be set on this shared multiplexed session?

Reply:
I was taking care of the broader `setSpan()` refactoring in a different PR - https://github.com/googleapis/java-spanner/pull/3016 . But now, will remove `span` from the constructor in this PR itself since its a small change.